### PR TITLE
HLT updates on top of CMSSW_7_4_3 (incl. B-Field from DB)

### PR DESCRIPTION
**Pull request by perrotta**
This PR allows the B-Field in HLT to be configured from the DB (and, in particular, it can be configured also for B different from the nominal one, i.e. B=0: @namapane ); dumps a new LowPU development menu; removes the recently added dumps of the two v2 version of the Frozen menus; integrates in the development HLT menus the other latest fixes and additions for the collision setups.

The Frozen menus v1 used for the MC production are not touched.

In addition, it merges the following PR's, needed for the HLT development, queued for 74X but not yet in the IB's:
- #9115 Adding L1T lowPU menu (74X)
- #9136: Bug fix of VBF Hbb code (74X)
- #9270:  FastSim Treatment in HLT tests (74X)

A similar PR for 75X is in preparation: yes, I know, I should have submitted the 75X version first. But 74X is the release used for HLT development, and I'm pushing this PR now just in case someone wants to consider it for a possible 74X patch release meant to test also HLT in the productions with B=0 (see also https://hypernews.cern.ch/HyperNews/CMS/get/physics-validation/2404.html)

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by davidlange6** [comment]
so the tables insisted on last week are gone this week?

**Comment by perrotta** [comment]
@davidlange6 : yes, I removed  that tables as we said that we only wanted them temporarily for the rate studies with 743 (and you also requested to get rid of them soon, if I remember correctly). But I can put them back, if you prefer so.

File: HLTrigger/Configuration/test/runAll.csh [hunk 1]
```diff
@@ -14,7 +14,7 @@ ls -l OnLine*.py
 #
 #foreach gtag ( Data Mc )
 #  set GTAG = ` echo $gtag | tr "[a-z]" "[A-Z]" `
-#  foreach table ( FULL GRun 50nsGRun HIon PIon 25ns14e33_v1 50ns_5e33_v1 25ns14e33_v2 50ns_5e33_v2 Fake )
+#  foreach table ( FULL GRun 50nsGRun HIon PIon 25ns14e33_v1 50ns_5e33_v1 Fake )
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/test/runAll.csh]: Should add LowPU

**Comment by fwyzard** [comment]
On 27 May 2015 at 13:01, David Lange <notifications@github.com> wrote:

> so the tables insisted on last week are gone this week?


They should be back once they are finalised, which at least for the 50ns
case probably means once we deploy it online, few weeks from now.

.Andrea


-- 
Strategy is a system of expedients.

Field Marshall Helmuth Carl Bernard Graf von Moltke

**Comment by Martin-Grunewald** [comment]
Hi @perrotta 
I believ in confdb.py we should update to the V2 version of the 50ns L1 menu,
as currently it is V1:
```
      if self.config.type not in ('Fake',) :
        if '50ns' in self.config.type :
          self.data += """
# load 2015 Run-2 L1 Menu for 50ns
from L1Trigger.Configuration.customise_overwriteL1Menu import L1Menu_Collisions2015_50ns_v1 as loadL1Menu
process = loadL1Menu(process)
"""
```

**Comment by perrotta** [comment]
Thank you @Martin-Grunewald : updated confdb.py, and I also took the opportunity of a couple of other  adjusements (not relevant for the final menu, as the commented our line that you also noticed)

**Comment by Martin-Grunewald** [comment]
I only see the v1=>v2 change in the commit

**Comment by perrotta** [comment]
(Because the other commit, 26b83447ba00dd7b778fcee55fddd38478921b54, is before your comment...)

**Comment by Martin-Grunewald** [comment]
:)

**Comment by Martin-Grunewald** [comment]
please test

**Comment by davidlange6** [comment]
Sounds like what we had agreed the right timing was in the first place (in the initial PR). Too bad we were told they were finalized last week so that we all had extra work to do.

>They should be back once they are finalised, which at least for the 50ns
>case probably means once we deploy it online, few weeks from now.

File: L1Trigger/L1TCommon/python/customsPostLS1.py [hunk 2]
```diff
@@ -78,9 +78,19 @@ def customiseSimL1EmulatorForStage1(process):
...
+def customiseSimL1EmulatorForPostLS1_lowPU(process):
+    # move to the lowPU v2 L1 menu once the HLT has been updated accordingly
+    process = L1Menu_Collisions2015_lowPU_v2(process)
+    return process
+
 def customiseSimL1EmulatorForPostLS1_50ns(process):
-    # move to the 50ns v0 L1 menu once the HLT has been updated accordingly
-    process = L1Menu_Collisions2015_50ns_v1(process)
+    # move to the 50ns v2 L1 menu once the HLT has been updated accordingly
+    process = L1Menu_Collisions2015_50ns_v2(process)
```
Comment (davidlange6) [inline L1Trigger/L1TCommon/python/customsPostLS1.py]: does this change not change the MC config?

**Comment by davidlange6** [comment]
@perrotta - #9138 should be dealt with separately (as not the hold up the rest). Since nothing depends on it, just remove it from this PR.

File: L1Trigger/L1TCommon/python/customsPostLS1.py (hunk 2 above)
Comment (perrotta) [inline L1Trigger/L1TCommon/python/customsPostLS1.py]: Uhm... yes, indeed.
I will revert it in this 74X PR

**Comment by perrotta** [comment]
@davidlange6 : about #9138, why do you think it should hold all the rest?
That PR looks absolutely innocuous to me, and it is already merged in 75X and fully signed in 74X: I can revert it if you insist on it, but I don't see the reason.

**Comment by perrotta** [comment]
+1

**Comment by davidlange6** [comment]
> On May 27, 2015, at 9:38 PM, perrotta <notifications@github.com> wrote:
> 
> @davidlange6 : about #9138, why do you think it should hold all the rest?
> That PR looks absolutely innocuous to me, and it is already merged in 75X and fully signed in 74X: I can revert it if you insist on it, but I don't see the reason.

#9138 needs relval testing - so if its in 9292 I won’t take #9292. Seems wrong for the other items to wait for the one thing which was included in the PR for no reason

**Comment by civanch** [comment]
+1

File: L1Trigger/L1TCommon/python/customsPostLS1.py (hunk 2 above)
Comment (fwyzard) [inline L1Trigger/L1TCommon/python/customsPostLS1.py]: In principle it does change the MC.
In practice, it only adds new L1 seeds, so it should not be a problem:
```diff
--- L1Menu_Collisions2015_50nsGct_v1    2015-05-28 09:08:48.193133942 +0200
+++ L1Menu_Collisions2015_50nsGct_v2    2015-05-28 09:08:51.385133895 +0200
@@ -2,6 +2,10 @@
    1                                                       L1_AlwaysTrue                                                       L1_AlwaysTrue
    2                                                        L1_SingleEG2                                                        L1_SingleEG2
    3                                                      L1_SingleJet16                                                      L1_SingleJet16
+   4                                            L1_MinimumBiasHF1_AND_v1                                               L1_MinimumBiasHF1_AND
+   5                                                L1_MinimumBiasHF1_OR                                                L1_MinimumBiasHF1_OR
+   6                                               L1_MinimumBiasHF2_AND                                               L1_MinimumBiasHF2_AND
+   7                                                L1_MinimumBiasHF2_OR                                                L1_MinimumBiasHF2_OR
    8                                    L1_Mu16er_TauJet40erORCenJet72er                                    L1_Mu16er_TauJet40erORCenJet72er
    9                                                        L1_Mu12_EG10                                                        L1_Mu12_EG10
   10                                                         L1_Mu5_EG15                                                         L1_Mu5_EG15
@@ -22,6 +26,7 @@
   25                                                 L1_DoubleTauJet44er                                                 L1_DoubleTauJet44er
   26                                                        L1_DoubleMu0                                                        L1_DoubleMu0
   27                                                     L1_SingleMu30er                                                     L1_SingleMu30er
+  28                                                            L1_ETT15                                                            L1_ETT15
   29                                             L1_Mu3_JetC16_WdEtaPhi2                                             L1_Mu3_JetC16_WdEtaPhi2
   30                                             L1_Mu3_JetC52_WdEtaPhi2                                             L1_Mu3_JetC52_WdEtaPhi2
   31                                                    L1_EG25er_HTT125                                                    L1_EG25er_HTT125
@@ -52,6 +57,7 @@
   56                                                       L1_SingleMu16                                                       L1_SingleMu16
   57                                                       L1_SingleEG15                                                       L1_SingleEG15
   58                                                        L1_SingleMu5                                                        L1_SingleMu5
+  59                                                            L1_ETT40                                                            L1_ETT40
   60                                                     L1_SingleMu20er                                                     L1_SingleMu20er
   61                                                       L1_SingleMu12                                                       L1_SingleMu12
   62                                                       L1_SingleMu20                                                       L1_SingleMu20
@@ -86,8 +92,12 @@
   91                                                    L1_Mu5_DoubleEG5                                                    L1_Mu5_DoubleEG5
   92                                                 L1_DoubleEG6_HTT150                                                 L1_DoubleEG6_HTT150
   93                                              L1_QuadJetC36_TauJet52                                              L1_QuadJetC36_TauJet52
+  94                                                            L1_ETT60                                                            L1_ETT60
+  95                                                            L1_ETT90                                                            L1_ETT90
+  96                                                     L1_DoubleMuOpen                                                     L1_DoubleMuOpen
   97                                                        L1_TripleMu0                                                        L1_TripleMu0
   98                                                   L1_TripleMu_5_5_3                                                   L1_TripleMu_5_5_3
+  99                                                      L1_SingleJet12                                                      L1_SingleJet12
  100                                                 L1_TripleEG_14_10_8                                                 L1_TripleEG_14_10_8
  101                                                   L1_DoubleEG_15_10                                                   L1_DoubleEG_15_10
  102                                                   L1_DoubleEG_22_10                                                   L1_DoubleEG_22_10
@@ -105,6 +115,8 @@
  114                                         L1_DoubleMu0_Eta1p6_WdEta18                                         L1_DoubleMu0_Eta1p6_WdEta18
  115                                             L1_ETM60_NotJet52WdPhi2                                             L1_ETM60_NotJet52WdPhi2
  116                                             L1_ETM70_NotJet52WdPhi2                                             L1_ETM70_NotJet52WdPhi2
+ 117                                                           L1_ETT130                                                           L1_ETT130
+ 118                                                     L1_RomanPotsAND                                                     L1_RomanPotsAND
  119                                               L1_TripleJet_92_76_64                                               L1_TripleJet_92_76_64
  121                                                          L1_QuadMu0                                                          L1_QuadMu0
  122                                                     L1_SingleMu18er                                                     L1_SingleMu18er
```

Comment (fwyzard) [inline L1Trigger/L1TCommon/python/customsPostLS1.py]: Ah, I see that it has been reverted anyway.

**Comment by perrotta** [comment]
Ok, @davidlange6 : I reverted #9138, as you requested
The additional commit prepared by Martin in the meanwhile look quite innocuous, but let me re-run all our tests...
Andrea

**Comment by Martin-Grunewald** [comment]
please test

File: L1Trigger/L1TCommon/python/customsPostLS1.py (hunk 2 above)
Comment (perrotta) [inline L1Trigger/L1TCommon/python/customsPostLS1.py]: Yes, I know it shouldn't have changed anything, as the new seeds are  
never accessed by the paths in the frozen menus used by the MC  
production.

Anyhow, try to stay on the safe side and avoid reasons to delay  
further the inclusion of this PR.

For the same reason, I just reverted #9138, as it was requested by David.

Hope we are done now...
Andrea

Andrea Bocci <notifications@github.com> ha scritto:

**Comment by perrotta** [comment]
Uhm, I am trying to rebase to get rid of extra signatures, but git does not seem to notice it.
Event though I rebased on the latest CMSSW_7_4_X_2015-05-27-2300, git keeps comparing it with the previous IB: strange...

**Comment by perrotta** [comment]
+1

**Comment by deguio** [comment]
+1

**Pull request merged**